src/main/java/org/springframework/samples/petclinic/rest/VisitRestController.java

```diff
@@ -78,12 +78,13 @@
 	public ResponseEntity<Visit> addVisit(@RequestBody @Valid Visit visit, BindingResult bindingResult, UriComponentsBuilder ucBuilder){
 		BindingErrorsResponse errors = new BindingErrorsResponse();
 		HttpHeaders headers = new HttpHeaders();
-		if(bindingResult.hasErrors() || (visit == null) || (visit.getPet() == null)){
+		if(bindingResult.hasErrors() || (visit == null) || (visit.getPet() == null) || visit.getVet() == null){
 			errors.addAllErrors(bindingResult);
 			headers.add("errors", errors.toJSON());
 			return new ResponseEntity<Visit>(headers, HttpStatus.BAD_REQUEST);
 		}
-		this.clinicService.saveVisit(visit);
+		visit.setPaid(false);
```

Comment:
The business logic must be in the service layer -> as mentioned in the general comment.